src/base/utils/python.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Mike Tzou
+ * Copyright (C) 2006  Christophe Dumez <chris@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <QString>
+
+namespace Utils
+{
+    namespace Python
+    {
+        int majorVersion();
```

Comment:
Wouldn't exporting the PythonInfo class from here by cleaner? E.g. 
```c++
ExternalExecutable<qint8, 3, 1> python()`
```

Reply:
Giving your answer to this below, the file might very well shrink to a single function. Maybe there is no reason to create this file here? What do you think about moving the implementations into `src/nase/utils/misc_python.cpp` and leaving declarations as they are?

Reply:
I think it will have more than 1 public function eventually, at least there should be a function for python version and an command executor function and maybe something else I cannot foresee.

Reply:
OK, but it would be natural to pack both these function into a class (for example, `ExternalExecutable` from my comment above). We use other external binaries, test their versions and launch them with arguments, so that, IMO, makes sense. And we get back to the "single function header" question.

Reply:
>We use other external binaries, test their versions and launch them with arguments, so that, IMO, makes sense. And we get back to the "single function header" question.

What if we need another "external binary"? wouldn't we still need at least 2 class declarations? maybe `<PL name>.h` is not the best choice, could be something like `foreignApp.h`... what I'm saying is it won't be a single function/class when there are multiple "external binary", isn't?

Reply:
> could be something like foreignApp.h

+1

> what I'm saying is it won't be a single function/class when there are multiple "external binary", isn't?

Functions are needed only for binaries not in the $PATH, but in general I agree with you.

Reply:
```c++
// candidate foreignApp.h layout
namespace Utils
{
  namespace ForeignApp
  {
    namespace Python
    {
      int majorVersion();
      QString executableName();
      QString fullVersion();
    }
  }
}
```

Reply:
ForeinApp**s**?